src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -313,3 +315,63 @@ def get_src_py_file_to_dest_py_file_map(
         # TODO: Does this case 5 need to be considered?
 
         return src_py_file_to_dest_py_file_map
+
+
+def _create_new_sql_file_at_path(parent_dir: Path, file_name: str) -> Optional[Path]:
+    """
+    Creates an empty SQL file at the same location specified as the parent directory, with file_name as part of the new file's name.
+    """
+    sql_file_name = f"__{file_name}_{uuid.uuid4().hex}.sql"
+    sql_file_path = Path(parent_dir, sql_file_name)
+    try:
+        f = open(sql_file_path, "x")
+        f.close()
+    except FileExistsError:
+        # It is unlikely that a user has created a file with this name, given the use of UUID.
+        cc.warning(
+            f"File {sql_file_path} already exists, cannot overwrite existing files."
+        )
+        cc.warning("Skipping creation of this file.")
+        return None
+    return sql_file_path
+
+
+def create_and_write_to_sql_files(dest_file_py_file_to_ddl_map: Dict[Path, str]):
+    """
+    Creates SQL files at predetermined location as specified in the input dictionary,
+    and writes the contents from the map to the new file.
+    """
+    dest_file_to_sql_file: Dict[Path, Path] = {}
+    for dest_file, all_ddl in dest_file_py_file_to_ddl_map.items():
+        if all_ddl is None or all_ddl.strip() == "":
```

Comment:
nit: weird, by this point I'd expect things to have been filtered out so that the generation logic doesn't have to deal with unexpected state. Also, your dict says Path -> str, so None is really unexpected here.

Reply:
Yes true, this is just an additional check, because this function can be generalized for any other create and write and we may not have control over what are the contents in the map outside the snowpark processor. I can remove it for now if you strongly feel one way.

I double checked about the type hint here, but at least in py3.11:
```
>>> my_dict: Dict[str, str] = {}
>>> my_dict["hello"] = None
>>> my_dict
{'hello': None}
```

Reply:
Python types are hints, they don't impact the behaviour at runtime because Python is very much dynamically typed. But using variables in a way that's inconsistent with type hints is just bad for understanding. Not to mention it's annoying to see the squigglies in an IDE :)